src/util.cpp

```diff
@@ -42,38 +42,54 @@ std::string generateUniqueName()
 
 } // namespace
 
-::mstch::node wrapYAMLNode(const YAML::Node &node, ScalarConversionFn fn)
+::mstch::node wrapYAMLNode(const YAML::Node &node, ScalarConversionFn fn,
+                           bool markLast)
 {
-    switch (node.Type())
-    {
-        case YAML::NodeType::Scalar:
-            return fn ? fn(node) : node.as<std::string>();
-        case YAML::NodeType::Sequence:
-        {
-            ::mstch::array context;
-            for (YAML::const_iterator it = node.begin(); it != node.end(); ++it)
-            {
-                const YAML::Node &value = *it;
-                context.emplace_back(wrapYAMLNode(value));
-            }
-            return context;
-        }
-        case YAML::NodeType::Map:
-        {
-            ::mstch::map context;
-            for (YAML::const_iterator it = node.begin(); it != node.end(); ++it)
-            {
-                const std::string name = it->first.as<std::string>();
-                const YAML::Node &value = it->second;
-                context[name] = wrapYAMLNode(value);
-            }
-            return context;
-        }
-        case YAML::NodeType::Undefined:
-        case YAML::NodeType::Null:
-        default:
-            return nullptr;
-    }
+  switch (node.Type())
+  {
+      case YAML::NodeType::Scalar:
+      {
+          return fn ? fn(node) : node.as<std::string>();
+      }
+      case YAML::NodeType::Sequence:
+      {
+          ::mstch::array context;
+
+          for (YAML::const_iterator it = node.begin(); it != node.end(); ++it)
+          {
+              const YAML::Node &value = *it;
+              context.emplace_back(wrapYAMLNode(value, nullptr, markLast));
```

Comment:
Actually, this should probably not be recursive, right?  If we have a sequence of sequences, the nested ones should not _also_ have the `{{last}}` added.
```suggestion
              context.emplace_back(wrapYAMLNode(value));
```